src/Stratis.Features.SQLiteWalletRepository/External/ScriptDestinationReader.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using NBitcoin.DataEncoders;
+using Stratis.Bitcoin.Interfaces;
+
+namespace Stratis.Features.SQLiteWalletRepository.External
+{
+    public interface IScriptDestinationReader : IScriptAddressReader
+    {
+        IEnumerable<TxDestination> GetDestinationFromScriptPubKey(Network network, Script script);
+    }
+
+    public class ScriptDestinationReader : IScriptAddressReader
+    {
+        private readonly IScriptAddressReader scriptAddressReader;
+
+        public ScriptDestinationReader(IScriptAddressReader scriptAddressReader)
+        {
+            this.scriptAddressReader = scriptAddressReader;
+        }
+
+        public string GetAddressFromScriptPubKey(Network network, Script script)
+        {
+            return this.scriptAddressReader.GetAddressFromScriptPubKey(network, script);
+        }
+
+        public virtual IEnumerable<TxDestination> GetDestinationFromScriptPubKey(Network network, Script redeemScript)
```

Comment:
A redeemScript is a P2SH concept, since the wallet is HD it is effectively P2PKH only (with P2PK support glued on). So the naming here is misleading.

Reply:
Yes, I am using "redeem script" in the sense of `TxOut.ScriptPubKey` - the script that needs to be extended with ScriptSig to redeem the funds.